Week6/news/src/app/news.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from "@angular/core";
+import { newsApiEndPoint } from "./Config/Endpoint";
+import { HttpClient } from "@angular/common/http";
+import { Observable } from "rxjs";
+
+@Injectable({
+  providedIn: "root"
+})
+export class NewsService {
+  public sourceName = "bbc-news";
+  public articles = [];
+  constructor(private http: HttpClient) {
+    this.getAuthorsList(this.sourceName).then(data => (this.articles = data));
+  }
+
+  getChannels(): Observable<any> {
+    return this.http.get("https://newsapi.org/v1/sources");
+  }
+
+  getAuthorsList(channelId) {
+    let channelAuthorEndpoint = newsApiEndPoint.articlesEndPoint;
+    const url = `${channelAuthorEndpoint}source=${channelId}&apiKey=${newsApiEndPoint.key}`;
+    return fetch(url)
```

Comment:
why fetch here instead of httpClient?

Reply:
no specific reason , i already used httpclient for another method .  So just copied code from previous task .